packages/inferno-animation/src/utils.ts

```diff
@@ -0,0 +1,206 @@
+declare global {
+  // Setting `window.__DEBUG_ANIMATIONS__ = true;` disables animation timeouts
+  // allowing breakpoints in animations for debugging.
+  // https://mariusschulz.com/blog/declaring-global-variables-in-typescript
+  var __DEBUG_ANIMATIONS__: Boolean;
+}
+
+export function addClassName (node: HTMLElement, className: string) {
+  if (className) {
+    const tmp = className.split(' ')
+    for (let i=0; i < tmp.length; i++) {
+      node.classList.add(tmp[i])
+    }
+  }
+}
+
+export function removeClassName (node: HTMLElement, className: string) {
+  if (className) {
+    const tmp = className.split(' ')
+    for (let i=0; i < tmp.length; i++) {
+      node.classList.remove(tmp[i])
+    }
+  }
+}
+
+export function forceReflow () {
+  return document.body.clientHeight;
+}
+
+export function setDisplay(node: HTMLElement, value?: string) {
+  var oldVal = node.style.getPropertyValue('display')
+  if (oldVal !== value) {
+    if (value !== undefined) {
+      node.style.setProperty('display', value)
+    }
+    else {
+      node.style.removeProperty('display')
+      _cleanStyle(node);
+    }
+  }
+  return oldVal
+}
+
+function _cleanStyle(node: HTMLElement) {
+  if (!node.style) {
+    // https://developer.mozilla.org/en-US/docs/Web/API/Element/removeAttribute
+    node.removeAttribute('style')
+  }
+}
+
+export function getDimensions(node: HTMLElement) {
+  var tmpDisplay = node.style.getPropertyValue('display')
+  var isDisplayNone = window.getComputedStyle(node).getPropertyValue('display') === 'none'
+  if (isDisplayNone) {
+    node.style.setProperty('display', 'block')
+  }
+
+  var tmp = node.getBoundingClientRect()
+
+  if (isDisplayNone) {
+    // node.style.display = tmpDisplay
+    node.style.setProperty('display', tmpDisplay)
+    _cleanStyle(node)
+  }
+
+  return {
+    width: tmp.width,
+    height: tmp.height
+  }
+}
+
+export function setDimensions(node: HTMLElement, width: number, height: number) {
+  node.style.width = width + 'px'
+  node.style.height = height + 'px'
+}
+
+export function clearDimensions(node: HTMLElement) {
+  node.style.width = node.style.height = ''
+}
+
+function _getMaxTransitionDuration (/* add nodes as args*/) {
+  let nrofTransitions = 0
+  let maxDuration = 0
+  for (let i=0; i < arguments.length; i++) {
+    const node = arguments[i]
+    if (!node) continue
+
+    const cs = window.getComputedStyle(node)
+    const dur = cs.getPropertyValue('transition-duration').split(',')
+    const del = cs.getPropertyValue('transition-delay').split(',')
+    const props = cs.getPropertyValue('transition-property').split(',').map((prop) => prop.trim())
+    props.forEach((prop) => {
```

Comment:
this should be converted to for - loop to avoid creating unnecessary function scopes especially because its inside another loop

Reply:
Will fix